python/tests/expressions/test_expressions_base.py

```diff
@@ -104,17 +109,95 @@ def visit_test_expression_b(self) -> List:
 
 
 @base.visit.register(ExpressionA)
-def _(obj: ExpressionA, visitor: BooleanExpressionVisitor) -> List:
+def _(obj: ExpressionA, visitor: FooBooleanExpressionVisitor) -> List:
     """Visit a ExpressionA with a BooleanExpressionVisitor"""
     return visitor.visit_test_expression_a()
 
 
 @base.visit.register(ExpressionB)
-def _(obj: ExpressionB, visitor: BooleanExpressionVisitor) -> List:
+def _(obj: ExpressionB, visitor: FooBooleanExpressionVisitor) -> List:
     """Visit a ExpressionB with a BooleanExpressionVisitor"""
     return visitor.visit_test_expression_b()
 
 
+class FooBoundBooleanExpressionVisitor(base.BoundBooleanExpressionVisitor[List]):
```

Comment:
Nit: Rather than `Foo` it would be better to name this `TestVisitor` or something.

Reply:
I originally had it as `TestBoundBooleanExpressionVisitor` but since classes that start with `Test` is a pytest convention, the test suite was getting confused. How about `ExampleVisitor`?